src/WorkflowFunction.ts

```diff
@@ -0,0 +1,214 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import {
+  WebClient,
+  FunctionsCompleteErrorResponse,
+  FunctionsCompleteSuccessResponse,
+} from '@slack/web-api';
+import {
+  Middleware,
+  AllMiddlewareArgs,
+  AnyMiddlewareArgs,
+  SlackEventMiddlewareArgs,
+  Context,
+} from './types';
+import processMiddleware from './middleware/process';
+import { WorkflowFunctionInitializationError } from './errors';
+
+/** Interfaces */
+
+export interface FunctionCompleteArguments {
+  outputs?: {
+    [key: string]: any;
+  };
+}
+
+export interface FunctionFailArguments {
+  error: string;
+}
+
+export interface FunctionCompleteFn {
+  (params?: FunctionCompleteArguments): Promise<FunctionsCompleteSuccessResponse>;
+}
+
+export interface FunctionFailFn {
+  (params: FunctionFailArguments): Promise<FunctionsCompleteErrorResponse>;
+}
+
+export interface WorkflowFunctionExecuteMiddlewareArgs extends SlackEventMiddlewareArgs<'function_executed'> {
+  complete: FunctionCompleteFn;
+  fail: FunctionFailFn;
+}
+
+/** Types */
+
+export type SlackWorkflowFunctionMiddlewareArgs = WorkflowFunctionExecuteMiddlewareArgs;
```

Comment:
I'd like for us to review these types and only export the ones that we really need in other files within the source. I have one eye on our types everywhere in our node SDKs, and it's getting messy. One way to try to control the type mess is by only exporting what you actually need throughout the project - this helps signal which types can be kept 'local' to a file which helps to manage that complexity.

This type, at first glance, might be one type that does not need to exist, since it just re-exports an already-existing interface? Not sure, but we should review the types before merging this PR.

Reply:
Again, carry over from the `WorkflowStep` feature. I think we saw developer requests to make available some of these, but I can't say for sure which. We can always wait for those requests to come in if they apply here, too.

I've combed this file and only exported that which is used by other files.